### PR TITLE
[SPARK-34894][CORE] Use 'io.connectionTimeout' as a hint instead of 'spark.network.timeout' for lost connections

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

Currently, when a connection for TransportClient is marked as idled and closed, we suggest users adjust `spark.network.timeout` for all transport modules. As a lot of timeout configs will fallback to the `spark.network.timeout`, this could be a piece of overkill advice, we should give a more targeted one with `spark.${moduleName}.io.connectionTimeout`

### Why are the changes needed?

better advise for overloaded network traffic cases

### Does this PR introduce _any_ user-facing change?

yes, when a connection is zombied and closed by spark internally, users can use a more targeted config to tune their jobs
### How was this patch tested?

Just log and doc. Passing Jenkins and GA

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @srowen @cloud-fan @maropu, Please have a look at this PR, thanks

**Review by cloud-fan — APPROVED** [review]

File: docs/configuration.md [hunk 1]
```diff
@@ -1997,6 +2007,16 @@ Apart from these, the following properties are also available, and may be useful
   </td>
   <td>3.0.0</td>
 </tr>
+<tr>
+  <td><code>spark.rpc.io.connectionTimeout</code></td>
```
Comment (HyukjinKwon) [inline docs/configuration.md:2011]: what about `spark.files.io.connectionTimeout`?

**Comment by HyukjinKwon** [comment]
cc @Ngone51 FYI

**Review by HyukjinKwon — APPROVED** [review]

**Review by srowen — APPROVED** [review]

**Review by Ngone51 — APPROVED** [review]

**Comment by yaooqinn** [comment]
retest this please

**Comment by mridulm** [comment]
LGTM

**Comment by yaooqinn** [comment]
thank you all, merging to master

**Pull request closed without merging**